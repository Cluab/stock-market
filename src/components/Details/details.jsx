import { PropTypes } from 'prop-types';
import './details.scss';

function Details({ props, symbol }) {
  const stocks = props;
  let stock;
  stocks.forEach((stocked) => {
    if (stocked[0].symbol === symbol) { stock = stocked; }
  });

  const {
    revenue, ebitda, operatingIncome, operatingIncomeRatio,
    operatingExpenses, date, costAndExpenses, costOfRevenue, grossProfit, grossProfitRatio, eps,
  } = stock[0];
  return (
    <>
      <div className="main-comp-section">
        <span className="fa-solid fa-money-bill-trend-up" />
        <div className="main-comp-name">
          <h2>{symbol}</h2>
          <span>
            {revenue}
            /
            Revenue
          </span>
        </div>
      </div>
      <div className="main-page-bar">
        {symbol}
        &nbsp;
        BREAKDOWN
        &nbsp;-
        &nbsp;
        {date}
      </div>
      <div>
        <ul className="comp-details">
          <li>
            <p>revenue</p>
            {' '}
            <span data-testid="test-revenue">{revenue}</span>
          </li>
          <li>
            <p>costOfRevenue</p>
            {' '}
            <span>{costOfRevenue}</span>
          </li>
          <li>
            <p>costAndExpenses</p>
            {' '}
            <span>{costAndExpenses}</span>
          </li>

          <li>
            <p>ebitda</p>
            {' '}
            <span>{ebitda}</span>
          </li>
          <li>
            <p>grossProfit</p>
            {' '}
            <span>{grossProfit}</span>
          </li>
          <li>
            <p>grossProfitRatio</p>
            {' '}
            <span>{grossProfitRatio}</span>
          </li>
          <li>
            <p>operatingExpenses</p>
            {' '}
            <span>{operatingExpenses}</span>
          </li>
          <li>
            <p>operatingIncome</p>
            {' '}
            <span>{operatingIncome}</span>
          </li>
          <li>
            <p>Earnings per share</p>
            {' '}
            <span data-testid="test-eps">{eps}</span>
          </li>
          <li>
            <p>OIR</p>
            {' '}
            <span>{operatingIncomeRatio}</span>
          </li>
        </ul>
      </div>
    </>
  );
}

Details.propTypes = {
  props: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.object)).isRequired,
  symbol: PropTypes.string.isRequired,
};
export default Details;
