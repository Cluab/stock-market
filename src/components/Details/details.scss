.comp-details {
  display: flex;
  flex-direction: column;

  & > li {
    font-weight: 400;
    padding: 20px;
    display: flex;
    flex-direction: row;
    color: white;
    justify-content: space-between;
    font-family: 'Roboto', sans-serif;

    & > i {
      font-size: 5rem;
      color: rgb(176, 57, 103);
    }

    & > h3 {
      font-size: 1.3rem;
    }

    & > p {
      font-size: 1rem;
    }
  }

  & > li:nth-child(odd) {
    background-color: rgb(234, 76, 138);
  }

  & > li:nth-child(even) {
    background-color: rgb(207, 66, 120);
  }
}
