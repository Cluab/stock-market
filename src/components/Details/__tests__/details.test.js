import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import Details from '../Details';
import valueReducer from '../../../redux/companies/companies';

const store = configureStore({ reducer: { stock: valueReducer } });
const stocks = [[{
  symbol: 'JPM',
  date: '2021-12-31',
  revenue: 121685000000,
  costOfRevenue: 0,
  costAndExpenses: 0,
  ebitda: 73047000000,
  grossProfit: 0,
  grossProfitRatio: 0,
  operatingExpenses: 0,
  operatingIncome: 0,
  eps: 15.39,
  operatingIncomeRatio: 0,
}]];

describe('companies component test cases', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Details props={stocks} symbol="JPM" />
          </Router>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match with input data ', () => {
    render(
      <Provider store={store}>
        <Router>
          <Details props={stocks} symbol="JPM" />
        </Router>
      </Provider>,
    );
    const stockSymbol = screen.getByTestId('test-eps').textContent;
    const stockRevenue = screen.getByTestId('test-revenue').textContent;

    expect(stockSymbol).toBe(String(stocks[0][0].eps));
    expect(stockRevenue).toBe(String(stocks[0][0].revenue));
  });
});
