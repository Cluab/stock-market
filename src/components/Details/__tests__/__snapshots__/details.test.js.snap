// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`companies component test cases should match with snapshot 1`] = `
Array [
  <div
    className="main-comp-section"
  >
    <span
      className="fa-solid fa-money-bill-trend-up"
    />
    <div
      className="main-comp-name"
    >
      <h2>
        JPM
      </h2>
      <span>
        121685000000
        / Revenue
      </span>
    </div>
  </div>,
  <div
    className="main-page-bar"
  >
    JPM
      BREAKDOWN  -  
    2021-12-31
  </div>,
  <div>
    <ul
      className="comp-details"
    >
      <li>
        <p>
          revenue
        </p>
         
        <span
          data-testid="test-revenue"
        >
          121685000000
        </span>
      </li>
      <li>
        <p>
          costOfRevenue
        </p>
         
        <span>
          0
        </span>
      </li>
      <li>
        <p>
          costAndExpenses
        </p>
         
        <span>
          0
        </span>
      </li>
      <li>
        <p>
          ebitda
        </p>
         
        <span>
          73047000000
        </span>
      </li>
      <li>
        <p>
          grossProfit
        </p>
         
        <span>
          0
        </span>
      </li>
      <li>
        <p>
          grossProfitRatio
        </p>
         
        <span>
          0
        </span>
      </li>
      <li>
        <p>
          operatingExpenses
        </p>
         
        <span>
          0
        </span>
      </li>
      <li>
        <p>
          operatingIncome
        </p>
         
        <span>
          0
        </span>
      </li>
      <li>
        <p>
          Earnings per share
        </p>
         
        <span
          data-testid="test-eps"
        >
          15.39
        </span>
      </li>
      <li>
        <p>
          OIR
        </p>
         
        <span>
          0
        </span>
      </li>
    </ul>
  </div>,
]
`;
